### PR TITLE
feat: timeline cell

### PR DESCRIPTION
**Pull request by nuanyang233**

### CODE REVIEW

File: src/types/dayjs-twitter.d.ts [hunk 1]
```diff
@@ -0,0 +1,10 @@
+declare module 'dayjs-twitter' {
```
Comment (guanbinrui) [inline src/types/dayjs-twitter.d.ts:1]: it should include in tsconfig.json

File: src/providers/lens/SocialMedia.ts [hunk 2]
```diff
@@ -263,20 +421,24 @@ export class LensSocialMedia implements Provider {
...
+            limit: LimitType.TwentyFive,
         });
 
+        const data = await Promise.allSettled(result.items.map((item) => this.formatPost(item)));
+
         return {
             indicator: indicator?.cursor,
             nextIndicator: result.pageInfo.next,
-            data: await Promise.all(result.items.map((item) => this.formatPost(item))),
+            data: compact(data.flatMap((x) => (x.status === 'fulfilled' ? x.value : null))),
```
Comment (guanbinrui) [inline src/providers/lens/SocialMedia.ts:441]: We could use createIndicator to compose the response object.

File: src/components/Posts/index.tsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+        queryFn: async ({ pageParam }) => {
+            return LensSocialMediaProvider.discoverPosts(createIndicator(undefined, pageParam));
+        },
+        initialPageParam: '',
+        getNextPageParam: (lastPage) => lastPage.nextIndicator,
+    });
+
+    return data.pages
+        .map((x) => x.data)
```
Comment (guanbinrui) [inline src/components/Posts/index.tsx:24]: flatMap

File: src/components/Posts/PostHeader.tsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
+import { Post } from '@/providers/types/SocialMedia';
+import { memo } from 'react';
+import { Image } from '@/components/Image';
+import More from '../../../public/svg/more.svg';
```
Comment (guanbinrui) [inline src/components/Posts/PostHeader.tsx:4]: Put inline images in src/assets/, public/ for publicly accessible contents.

File: src/components/Markup/MarkupLink/Hashtag.tsx [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+import { memo } from 'react';
+
+export const Hashtag = memo<MarkupLinkProps>(function Hashtag({ title }) {
+    if (!title) return null;
+
+    const tag = title.slice(1).toLowerCase();
+
+    return (
+        <span className="inline-flex">
```
Comment (guanbinrui) [inline src/components/Markup/MarkupLink/Hashtag.tsx:14]: This node seems redundant.

File: src/components/Markup/MarkupLink/ExternalLink.tsx [hunk 6]
```diff
@@ -0,0 +1,25 @@
+import { MarkupLinkProps } from '@/components/Markup/MarkupLink';
+import { formatUrl } from '@/helpers/formatUrl';
+import { useMounted } from '@/hooks/useMounted';
+import Link from 'next/link';
+import { memo } from 'react';
+
+export const ExternalLink = memo<MarkupLinkProps>(function ExternalLink({ title }) {
+    const isMounted = useMounted();
```
Comment (guanbinrui) [inline src/components/Markup/MarkupLink/ExternalLink.tsx:8]: isMounted not used.

File: src/components/Image.tsx [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+        [imageLoadFailed, setImageLoadFailed, onError],
+    );
+
+    useEffect(() => {
+        setImageLoadFailed(false);
+    }, [props.src]);
+
+    // TODO: replace failed fallback image
+    return (
+        <img
```
[the remark this answers is not in the document]
Comment (nuanyang233) [reply src/components/Image.tsx:31]: `next/image` relies on defining all the image domains in advance in `next.config`, but the images in the rendering timeline are third-party and we can't predict which domains will be available, so we'll use `img` tags instead.

**Comment by nuanyang233** [comment]
Too many conflicts and I will reopen a new pr.

**Pull request closed without merging**